engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorContextImpl.java

```diff
@@ -107,6 +111,12 @@ public HibernateValidatorContext clockProvider(ClockProvider clockProvider) {
 	}
 
 	@Override
+	public HibernateValidatorContext addValueExtractor(ValueExtractor<?> extractor) {
+		valueExtractors.add( extractor );
```

Comment:
Yes, the public methods would add VEs but we would convert them to VED as early as possible to throw an error as early as possible and use as much as possible VED internally.